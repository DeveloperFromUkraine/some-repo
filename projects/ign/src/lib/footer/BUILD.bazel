load("@angular//:index.bzl", "ng_module")
load("@io_bazel_rules_sass//:defs.bzl", "sass_binary")

package(default_visibility = ["//visibility:public"])

sass_binary(
    name = "style",
    src = "footer.scss",
)

ng_module(
    name = "footer-module",
    srcs = [
        "footer.module.ts",
        "footer.component.ts",
    ],
    assets = [":footer.css", "footer.html"],
    tsconfig = "//:tsconfig.json",
    deps = [
        "@npm//tslib",
        "@npm//@types",
        "@angular//packages/core",
        "@angular//packages/common",
    ]
)
