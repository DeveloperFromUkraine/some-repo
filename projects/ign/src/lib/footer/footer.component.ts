import { Component } from '@angular/core';

@Component({
  selector: 'ign-footer',
  templateUrl: './footer.html',
  styleUrls: ['./footer.css'],
})
export class FooterComponent {}
