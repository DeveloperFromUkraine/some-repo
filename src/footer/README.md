## Use
    Footers are used for information to be located at the bottom of the page. The footer component is a stylized version of a footer. 

    ### Implementation
        * Footers will be placed at the the bottom of containers. 
        * Footers should contain:
            * Links
            * Icons
            * Contact Information
            * Copyrights
        * Footer components should contain hyperlinks to outside sources pertaining to the company or other pages   within the same user experience.
        * Footers should not contain descriptive content.
        * Footers should contain logos and icons in lieu of text.

## API
    ### Selector
        * ign-footer
    
    ### Export As
        * FooterComponent

## Implementation HTML
    I am not a footer.
    <ign-footer>I am a footer</ign-footer>

## Implementation TS
    None