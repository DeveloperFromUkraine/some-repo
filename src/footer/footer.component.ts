import { Component } from '@angular/core';

@Component({
  selector: 'ign-footer',
  templateUrl: './footer.html',
  styleUrls: ['./footer.scss'],
})
export class FooterComponent {}
