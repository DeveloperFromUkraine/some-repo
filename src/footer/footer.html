<footer class="footer" role="contentinfo">
  <p class="footer-text">
    <ng-content></ng-content>
  </p>
</footer>