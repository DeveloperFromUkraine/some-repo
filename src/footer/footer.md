*** HTML ***
# Footer

## Use
 Footer components are designed to contain information pertinent to the company, user, or other forms of
    non-content related navigation.

## Sample
<mat-tab-group>
    <mat-tab label="Component Sample">
        <div class="tab-height">
            I am not a footer.
            <ign-footer>I am a footer</ign-footer>
        </div></mat-tab>
    <mat-tab label="HTML"><div class="tab-height">
        <table style="width:100%">
            &lt;ign-footer&gt;I am a footer&lt;/ign-footer&gt;
        </table>
    </div></mat-tab>
</mat-tab-group>

## Content

* Footer components should contain hyperlinks to outside sources pertaining to the company or other pages within
the same user experience.
* Footers should not contain descriptive content.
* Footers should contain logos and icons in lieu of text.


## Style
* Footers must be placed along the bottom of the page.

*** TypeScript *** 

*** ExternalTypeScript ***

*** Imports ***