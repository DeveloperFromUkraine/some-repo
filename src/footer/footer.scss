@import '~ignite-common/styles/ignite';

$text-secondary-font-size: 12px;
$text-secondary-font-weight: 400;

:host {
  width: 100%;
}

.footer {
  width: 100%;
  margin: 0 auto;
  padding: 1rem;
}
.footer-text {
  font-size: $text-secondary-font-size;
  font-weight: $text-secondary-font-weight;
  color: ign-color(text-secondary);
}