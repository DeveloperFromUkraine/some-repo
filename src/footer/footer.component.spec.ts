import { ComponentFixture, TestBed } from "@angular/core/testing";
import { FooterComponent } from "./footer.component";
import { ComponentTest } from "../../test/test-bed/component";

describe('FooterComponent', () => {
    let fixture: ComponentFixture<FooterComponent>;

    beforeEach(async () => {
        await ComponentTest.createTestBed([], [FooterComponent]);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(FooterComponent);
    });

    it('should match snapshot', () => {
        expect(fixture).toMatchSnapshot();
    });
});