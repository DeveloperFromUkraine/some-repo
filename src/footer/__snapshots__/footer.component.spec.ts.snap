// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<ign-footer>
  <footer
    class="footer"
    role="contentinfo"
  >
    <p
      class="footer-text"
    />
  </footer>
</ign-footer>
`;
